packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -338,27 +338,24 @@ const drawCommits = (svg, commits, modifyGraph) => {
 };
 
 /**
- * Detect if there are other commits between commit1's x-position and commit2's x-position on the
- * same branch as commit2.
+ * Detect if there are commits
+ * between commitA's x-position
+ * and commitB's x-position on the
+ * same branch as commitA
  *
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param allCommits
- * @returns {boolean} If there are commits between commit1's x-position and commit2's x-position
+ * @returns {boolean}
+ * If there are commits between
+ * commitA's x-position
+ * and commitB's x-position
  */
-const hasOverlappingCommits = (commit1, commit2, allCommits) => {
-  // Find commits on the same branch as commit2
-  const keys = Object.keys(allCommits);
-  const overlappingComits = keys.filter((key) => {
-    return (
-      allCommits[key].branch === commit2.branch &&
-      allCommits[key].seq > commit1.seq &&
-      allCommits[key].seq < commit2.seq
-    );
-  });
-
-  return overlappingComits.length > 0;
-};
+const hasOverlappingCommits = (commitA, commitB, allCommits) =>
+  Object.entries(allCommits).some(
+    (commitX) =>
+      commitX.branch === commitA.branch && commitX.seq > commitA.seq && commitX.seq < commitB.seq
```

Comment:
I do not know if it is intended change, or is there any difference which one to use, I would have to debug the code to say that, but previously it was
```
commit.branch === commit2.branch
```
and now it has turned into
```
commit.branch === commitA.branch
```
which basically means
```
commit.branch === commit1.branch // used to be commit2
```

Reply:
Thanks for comments so far -- great feedback.

That change to the choice of commit branch was deliberate -- here is [the commit in which I correct from `commit2` to `commit1`](https://github.com/mermaid-js/mermaid/pull/4927/commits/fe0555f9d7f1fa85728ca3a04a92d453332559e3) from before I renamed the variables. It solves the particular bug I was focused on.

However, I did find there may be another (slightly separate) bug which is live right now but that this PR does not address, but it would be part of the same line of code. So, while I'm checking your other feedback, I'll take a look to see if it makes sense to address it here.

Reply:
I rewrote (and renamed) this function to make it clearer as well as fixing the issues. Hope this makes more sense now.